### PR TITLE
Add katago

### PR DESCRIPTION
**Pull request by hadim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hadim** [comment]
@conda-forge/help-c-cpp, can you review, please?

On OSX during linking, there is this error:

```
ld: library not found for -latomic
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
```

`libatomic` seems to be shipped by `libgcc-ng` but it's not available for OSX.

**Comment by hadim** [comment]
Thanks to @chrisburr for pointing out the CMake config that might be buggy for OSX. It seems to be fixed but I see a new error on OSX now. It's an `Illegal instruction` error when executing `katago runtests`. I will test disabling AVX2 on osx.

**Comment by hadim** [comment]
@chrisburr can you review please?

File: recipes/katago/LICENSE [hunk 1]
```diff
@@ -0,0 +1,35 @@
+The code in this repository currently relies on several other libraries, parts of libraries,
```
Comment (chrisburr) [inline recipes/katago/LICENSE:1]: I don't think this file is needed. If it's in the source tarball it'll be taken from there (the key in the `meta.yaml` is still needed).

File: recipes/katago/osx-cmake.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
+index 6f2a9b4c..54a9fc8f 100644
+--- a/cpp/CMakeLists.txt
++++ b/cpp/CMakeLists.txt
+@@ -418,7 +418,7 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang
+   else()
+     message(STATUS "Enabling Clang-specific build options.")
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnull-dereference -Wdangling-else")
+-    target_link_libraries(katago "atomic")
++    # target_link_libraries(katago "atomic")
```
Comment (chrisburr) [inline recipes/katago/osx-cmake.patch:10]: Have you opened an issue upstream about this?

Comment (hadim) [reply recipes/katago/osx-cmake.patch:10]: Xref with https://github.com/lightvector/KataGo/issues/690

**Comment by hadim** [comment]
Thank you @chrisburr

**Comment by hadim** [comment]
xref https://github.com/conda-forge/katago-feedstock/pull/1

**Pull request merged**